src/org/recompile/freej2me/Anbu.java

```diff
@@ -56,259 +56,268 @@
-			{
+			{
 				System.out.println("Unable to initialize SDL");
 				stop();
 			}
 
-			// Clear screen and draw coloured Background
-			// if(angle == 270) { SDL_CreateWindowAndRenderer(sourceHeight*windowScale, sourceWidth*windowScale, SDL_WINDOW_SHOWN, &mWindow, &mRenderer); }
-			// else {
-				// SDL_CreateWindowAndRenderer(sourceWidth*windowScale, sourceHeight*windowScale, SDL_WINDOW_SHOWN, &window, &renderer);
-			window = SDL_CreateWindow("FreeJ2ME-Plus - SDL", SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED, lcdWidth, lcdHeight, SDL_WINDOW_SHOWN);
-			renderer = SDL_CreateRenderer(window, -1, 0);
-
-			// }
-			// if(isFullscreen) { toggleFullscreen(); }
-			// SDL_SetRenderDrawColor(mRenderer, r, g, b, 255);
-			SDL_RenderClear(renderer);
-			SDL_RenderPresent(renderer);
-
-			// Set scaling properties
-			// SDL_SetHint(SDL_HINT_RENDER_SCALE_QUALITY, interpol.c_str());
-			// SDL_RenderSetLogicalSize(mRenderer, displayWidth, displayHeight);
+			window = SDL_CreateWindow("FreeJ2ME-Plus - SDL", SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED, lcdWidth*scaleFactor, lcdHeight*scaleFactor, SDL_WINDOW_SHOWN);
 			
-			// Create a mTexture where drawing can take place. Streaming for constant updates.
+			// Create a renderer, and a texture where drawing can take place, streaming for constant updates.
+			renderer = SDL_CreateRenderer(window, -1, 0);
 			texture = SDL_CreateTexture(renderer, SDL_PIXELFORMAT_RGB24, SDL_TEXTUREACCESS_STREAMING, lcdWidth, lcdHeight);
-
-			pixels = new Memory(lcdWidth * lcdHeight * 3);
-
+			pixels = new Memory(lcdWidth * lcdHeight * 3); 
+
+			/* 
+			 * Input reading should not be tied to the render logic, otherwise jars that only send new render data 
+			 * after an input is registered (Ex: JBenchmark 2 and some other jars that use Form UI) will get softlocked.
+			 */
+			keytimer = new Timer();
+			keytask = new TimerTask() {	public void run() { processEvents(); } };
+			keytimer.schedule(keytask, 0, 1);
+
 			SDL_JoystickEventState(SDL_ENABLE);
-		}
-
-		public void stop()
-		{
+
+			SDLInitialized = true;
+		}
+
+		public void stop()
+		{
 			SDL_DestroyTexture(texture);
-			System.exit(0);
-		}
-
+			System.exit(0);
+		}
+
 		public void paint()
-		{
-			processEvents();
-
-			int[] data;
-
-			data = Mobile.getPlatform().getLCD().getRGB(0, 0, lcdWidth, lcdHeight, null, 0, lcdWidth);
-			
-			int cb = 0;
+		{
+
+			/* 
+			 * Let's make resolution changes and adjust any relevant objects here, as it's right on the render path 
+			 * and it makes sure that the objects will be set correctly before being rendered.
+			 */
+			if(resolutionChanged) 
+			{
+				SDL_SetWindowSize(window, lcdWidth*scaleFactor , lcdHeight*scaleFactor);
+				SDL_DestroyRenderer(renderer);
+				renderer = SDL_CreateRenderer(window, -1, 0);
+				texture = SDL_CreateTexture(renderer, SDL_PIXELFORMAT_RGB24, SDL_TEXTUREACCESS_STREAMING, lcdWidth, lcdHeight);
+				sdl.pixels = new Memory(lcdWidth * lcdHeight * 3); 
```

Comment:
If this needed or part of the workaround? Only `SDL_SetWindowSize()` should be required, and the game I'm testing is crash on `SDL_DestroyRenderer()` here after a few seconds, possibly caused by a resolution change?

Reply:
Unfortunately DestroyRenderer is required, at least here, in order to even show something on the screen after booting. Still, for now that should only trigger right at boot unless the user is changing the screen res on the fly, which shouldn't be possible yet (scaling works fine from what i've tested, though it did also crash previously before i put that block inside 'paint()' to make sure it only got called before an actual render). Which game is that? I'll take a look at it and see what it might be trying to do.